Assets/Scripts/Entities/Enemy.cs

```diff
@@ -137,5 +146,15 @@ private void OnDrawGizmos()
             Gizmos.color = Color.yellow;
             Gizmos.DrawWireSphere(transform.position, aggravationRange);
         }
+
+        private void SetupEventSubscriptions()
+        {
+            _gameController.timer.OnPhaseChange.AddListener(SetWorldPhase);
```

Comment:
Do we need to remove the listener in the Die() function?

Reply:
The answer to your question, is yes. We do need to remove it. Will you add it?

Reply:
Sweet, done. Not tested though as I can't attack?

Reply:
Yeah attacking is in progress XD